File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/projectcookie/ProjectCookies.java
package ru.herooo.mylanguageweb.projectcookie;

public final class ProjectCookies {
    public static final ProjectCookie CUSTOMER_AUTH_KEY =
            new ProjectCookie("customer_auth_key", 7 * 24 * 60 * 60, false, true);

    public static final ProjectCookie CUSTOMER_ID =
            new ProjectCookie("customer_id", 7 * 24 * 60 * 60, false, false);

    public static final ProjectCookie WORKOUT_AUTH_KEY =
            new ProjectCookie("workout_auth_key", 24 * 60 * 60, false, true);
}

File: MyLanguageWeb/src/main/resources/static/js/classes/html/project_cookies.js
export class ProjectCookies {
    CUSTOMER_ID = new ProjectCookie(
        "customer_id",
        {
            'max-age': 7 * 24 * 60 * 60
        });
}

class ProjectCookie {
    #name;
    #options;

    constructor(name, options = {}) {
        this.#name = name;
        this.#options = options;
    }

    getValue() {
        let matches = document.cookie.match(new RegExp(
            "(?:^|; )" + this.#name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1') + "=([^;]*)"
        ));
        return matches ? decodeURIComponent(matches[1]) : undefined;
    }

    setValue(value) {
        this.#setCookie(this.#name, value, this.#options);
    }

    clearValue() {
        this.#setCookie(this.#name, "", {'max-age': -1});
    }

    #setCookie(name, value, options = {}) {
        options = {
            path: '/',
            // при необходимости добавьте другие значения по умолчанию
            ...options
        };

        let updatedCookie = encodeURIComponent(name) + "=" + encodeURIComponent(value);

        for (let optionKey in options) {
            updatedCookie += "; " + optionKey;
            let optionValue = options[optionKey];
            if (!optionValue) {
                updatedCookie += "=" + optionValue;
            }
        }

        document.cookie = updatedCookie;
    }
}